front-end-tools/src/components/InitComponent.tsx

```diff
@@ -0,0 +1,643 @@
+                            <br />
+                            <div>
+                                <b>Uncheck</b> this box, if you want to manually fill in a <code>module reference</code>
+                                , the <code>smart contract name</code>, or an <code>input parameter schema</code>.
+                            </div>
+                            <br />
+                            <div>
+                                <b>Uncheck</b> and <b>check</b> this box again, if you want to load a new module from
+                                step 1.
+                            </div>
+                        </Alert>
+                        <br />
+                    </>
+                )}
+
+                {moduleReferenceError && <Alert variant="danger"> Error: {moduleReferenceError}. </Alert>}
+
+                <Row>
+                    <Form.Group className="col-md-4 mb-3">
+                        <Form.Label> Module Reference</Form.Label>
+                        <Form.Control
+                            defaultValue="91225f9538ac2903466cc4ab07b6eb607a2cd349549f357dfdf4e6042dde0693"
+                            disabled={!!useModuleReferenceFromStep1}
+                            {...form.register('moduleReference', { required: true })}
+                            onChange={(e) => {
+                                const register = form.register('moduleReference', {
+                                    required: true,
+                                });
+
+                                register.onChange(e);
+
+                                setModuleReferenceLengthError(undefined);
+
+                                const moduleRef = form.getValues('moduleReference');
+
+                                if (moduleRef !== undefined && moduleRef.length !== 64) {
+                                    setModuleReferenceLengthError('Module reference has to be of length 64');
+                                }
+                            }}
+                        />
+                        <Form.Text />
+                        {form.formState.errors.moduleReference && (
+                            <Alert variant="info">Module reference is required </Alert>
+                        )}
+                    </Form.Group>
+
+                    {useModuleReferenceFromStep1 && displayContracts.length > 0 ? (
+                        <Form.Group className="col-md-4 mb-3">
+                            <Form.Label>Smart Contract Name</Form.Label>
+
+                            <Select
+                                options={displayContracts?.map((contract) => ({
+                                    value: contract,
+                                    label: contract,
+                                }))}
+                                onChange={(e) => {
+                                    form.setValue('smartContractName', e?.value);
+                                }}
+                            />
```

Comment:
I think it would  make sense to pre-select  the first one instead of nothing. In almost all cases there will be exactly one contract, so it's annoying to have to manually select it.